plane_wave.ipynb

```diff
@@ -0,0 +1,643 @@
+    "ax.legend()\n",
+    "ax.grid(True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "41da0745",
+   "metadata": {},
+   "source": [
+    "## 3.2 Animation of standing wave"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e7a23115",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "flag = 'cpx'  # 'cpx', re'\n",
+    "n_periods = 2\n",
+    "n_samples_per_period = 64\n",
+    "Ts = 2*np.pi / (w*n_samples_per_period)  # s, sampling period\n",
+    "\n",
+    "\n",
+    "def init():\n",
+    "    ax.set_xlabel('x / m')\n",
+    "    ax.set_ylabel(r'$\\frac{p}{pA}$, $\\frac{v}{pA}\\cdot\\rho_0 c$')\n",
+    "    ax.set_xlim(x[0], x[-1])\n",
+    "    ax.set_ylim(-2, 0)\n",
+    "    ax.set_xticks(np.linspace(-2*wave_length, 0, 5))\n",
+    "    ax.set_yticks(np.linspace(-2, 2, 5))\n",
+    "    ax.grid(True)\n",
+    "\n",
+    "\n",
+    "def update(frame):\n",
+    "    ax.clear()\n",
+    "    init()\n",
+    "\n",
+    "    if flag == 'cpx':\n",
+    "\n",
+    "        time_phasor = np.exp(+1j*w*Ts*frame)\n",
+    "        ax.plot(x, np.real(p_cpx/pA*time_phasor), 'C0', label='pressure')\n",
+    "        ax.plot(x, np.real(v_cpx/pA*rho0*c*time_phasor), 'C1', label='velocity')\n",
+    "\n",
+    "    elif flag == 're':\n",
+    "\n",
+    "        time_phasor_cos = np.cos(w*Ts*frame)\n",
+    "        time_phasor_sin = np.sin(w*Ts*frame)\n",
+    "        ax.plot(x, p_re/pA*time_phasor_cos, 'C0', label='pressure')\n",
+    "        ax.plot(x, v_re/pA*rho0*c*time_phasor_sin, 'C1', label='velocity')\n",
+    "\n",
+    "    ax.set_title('elapsed time {time:4.3f} ms'.format(time=Ts*frame*1000))\n",
+    "    ax.legend(loc='upper left')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "73eff736",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig, ax = plt.subplots(figsize=(7, 4), nrows=1, ncols=1)\n",
+    "update(0)\n",
+    "init()\n",
+    "\n",
+    "ani = FuncAnimation(fig, update,\n",
+    "                    frames=np.arange(n_periods*n_samples_per_period+4),\n",
+    "                    blit=True, interval=50,\n",
+    "                    repeat=False)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7720ba40",
+   "metadata": {},
+   "source": [
+    "<p xmlns:dct=\"http://purl.org/dc/terms/\">\n",
+    "  <a rel=\"license\"\n",
+    "     href=\"http://creativecommons.org/publicdomain/zero/1.0/\">\n",
+    "    <img src=\"http://i.creativecommons.org/p/zero/1.0/88x31.png\" style=\"border-style: none;\" alt=\"CC0\" />\n",
+    "  </a>\n",
+    "  <br />\n",
+    "  To the extent possible under law,\n",
+    "  <span rel=\"dct:publisher\" resource=\"[_:publisher]\">the person who associated CC0</span>\n",
+    "  with this work has waived all copyright and related or neighboring\n",
+    "    rights to this work. Author: <a href=\"https://github.com/fs446\">https://github.com/fs446</a>\n",
```

Comment:
If you want to include authorship information (which I specifically chose *not* to do, in order to lower the bar for contributions), you can use the tool at https://creativecommons.org/choose/zero/waiver. Mentioning your name here does not change the obligations of users (they still have none).

If you want to legally require people to mention your name when they reproduce/change the materials, you should use CC-BY instead.

Reply:
Thanks for this hint! I removed the autorship, such that we are consistent within this course. I only liked the idea, that interested people find autorship information directly in the notebook and don't need to trace this in the git repo (where this information is encoded anyway). However, I did not figure the implications of different licensing models. So making it most convenient for re-usage is nice to me too.